draft-httpbis-doh-preference-hints.md

```diff
@@ -177,6 +192,13 @@ Clients MUST NOT use the contents of the DoH-Preference header to impact how
 it resolves other domain names. Clients MUST ignore the DoH-Preference header
 in HTTP responses conveyed over a non-secure transport.
 
+If the DoH-Preference URI contains a host expressed as a host name rather than
+as an IP address and that host name is resolved via DoH, the DoH server might
+also specify a DoH-Preference header. This means that respecting the DoH server
+recommendation could result in an excessively long chain of DoH queries or a
+loop of DoH servers. Clients SHOULD be capable of detecting a loop or an
+excessively long chain of DoH servers and treat these conditions as a query
+failure.
```

Comment:
And for query failures, we fall back to native DNS, correct?

Reply:
Yes, this is indicated elsewhere in the document, I was trying to mimic that language so that would be clear.